crates/e2e-test-utils/src/lib.rs

```diff
@@ -165,7 +165,9 @@ where
         let NodeHandle { node, node_exit_future: _ } = NodeBuilder::new(node_config.clone())
             .testing_node(exec.clone())
             .with_types_and_provider::<N, BlockchainProvider<_>>()
-            .with_components(node.components_builder())
+            .with_components::<BuilderComponentsAdapter<FullNodeTypesAdapter<_, _, _>, _>>(
```

Comment:
this we don't want because this leaks into user code, forcing the user to provide this adapter mapping